x/mint/types/inflations.go

```diff
@@ -0,0 +1,65 @@
+package types
+
+import (
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// NewInflation returns a new Inflation object with the given denom, target_address
+// and inflation_rate
+func NewInflation(denom string, targetAddress string, inflationRate sdk.Dec) Inflation {
+	return Inflation{
+		Denom:         denom,
+		TargetAddress: targetAddress,
+		InflationRate: inflationRate,
+	}
+}
+
+func CalculateInflationPerBlock(inflation *Inflation, blocksPerYear uint64) (result sdk.Dec, err error) {
+	inflationPerBlockPlusOne, err := inflation.InflationRate.Add(sdk.OneDec()).ApproxRoot(blocksPerYear)
+	if err != nil {
+		panic(err)
```

Comment:
I would not panic here, just `return err`.
Panic would have sense only if this function would not return tuple with `err` in it.